src/main/java/com/epam/cdp/m2/hw2/aggregator/Java8Aggregator.java

```diff
@@ -0,0 +1,60 @@
+package com.epam.cdp.m2.hw2.aggregator;
+
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import javafx.util.Pair;
+
+public class Java8Aggregator implements Aggregator {
+
+    @Override
+    public int sum(List<Integer> numbers) {
+        return numbers.stream()
+                .mapToInt(Integer::new)
+                .sum();
+    }
+
+    @Override
+    public List<Pair<String, Long>> getMostFrequentWords(List<String> words, long limit) {
+        List<Pair<String, Long>> result = words.stream()
+                .map(String::toLowerCase)
+                .collect(Collectors.groupingBy(w -> w,
+                        Collectors.summingLong(w -> 1L)))
+                .entrySet().stream()
+                .map(es -> new Pair<>(es.getKey(), es.getValue()))
+                .sorted(Comparator.comparing(Pair::getValue)) // TODO: Figure out why `Comparator.comparing(Pair::getValue).reversed()` throws error
+//                 .sorted(Comparator.reverseOrder())
+//                .limit(limit)
+                .collect(Collectors.toList());
+
+        Collections.reverse(result);
+
+        return result.stream()
+                .limit(limit)
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public List<String> getDuplicates(List<String> words, long limit) {
+//        return words.stream()
+//                .filter(word -> Collections.frequency(words, word) > 1) // TODO: Is there possible to iterate here on mapped array?
```

Comment:
In this solution you should change words to contain uppercase words to calculate duplicates correctly.
Also at the and you should remove duplicates from result.
Something like this:
words.stream()
                .map(String::toUpperCase)
                .filter(word -> Collections.frequency(words.stream().map(String::toUpperCase).collect(Collectors.toList()), word) > 1) 
                .limit(limit)
                .collect(Collectors.toSet())
                .stream()
                .collect(Collectors.toList())
                
Your uncomment solution looks better.

Reply:
yeah, I agree